docs/DeveloperGuide.md

```diff
@@ -258,6 +262,47 @@ Through the logic in the object of `UserStatsCalculator`, necessary information
 
 A similar procedure is being employed by the `TopicalStatsCalculator` object to calculate the topic-level statistics for the user. The only difference between the objects of these two classes is that instead of iterating through all the topics available, the `TopicalStatsCalculator` object only deals with a particular topic at any point of time. By iterating through the questions of the single topic, it calculates statistics for the topic and returns it back to the `Stats` object, which will then pass them to the `Ui` object to display them to the user. As such, in order to display the user’s statistics for each and every topic, a loop is done in the `Stats` object to repeatedly calculate the topic-level information for all of the topics and displaying them concurrently. 
 
+#### 2.3.7. Implementation of Timer Feature
+
+E-Duke-8 has a timer feature for the quiz. The users are able to choose how much time (in seconds) they require to complete each question. 
+The timer would start as soon as the options are being printed out. 
+If the user has typed in a valid answer, the timer would stop and move on to the next question. 
+If the time is up, it will be regarded as an `IncompleteCommand` and it will be deemed as a wrong answer as well. 
+
+The diagram below only illustrates the implementation of the timer feature, it does not show the full sequence diagram of the `SingleTopicQuiz()` method. 
+
+![Timer_Sequence_Diagram](./images/Timer.png)
```

Comment:
Good effort with adding the implementation of timer feature. However, I think rather than constructing the `SingleTopicQuiz` object here you should be calling the `startQuiz()` method on an existing one instead. Also I noticed the `Question` object is not being used so it should be taken out. Furthermore I think if `QuizParser` is not an important part of the implementation of the timer it may also be excluded for brevity.

Reply:
Thank you for spotting! I have updated the diagram already

Reply:
Looks good